platform/darwin/src/MGLBackgroundStyleLayer.mm

```diff
@@ -92,7 +92,7 @@ - (void)setBackgroundColor:(MGLStyleValue<MGLColor *> *)backgroundColor {
 - (MGLStyleValue<MGLColor *> *)backgroundColor {
     MGLAssertStyleLayerIsValid();
 
-    auto propertyValue = self.rawLayer->getBackgroundColor() ?: self.rawLayer->getDefaultBackgroundColor();
+    static const auto propertyValue = self.rawLayer->getBackgroundColor() ?: self.rawLayer->getDefaultBackgroundColor();
```

Comment:
Well that doesn't make much sense does it? It stops the warning but causes other more serious issues, obviously. I'll keep looking.